docs/libvirt-pod-networking.md

```diff
@@ -0,0 +1,98 @@
+# Using CNI Interfaces and libvirt for networking
+
+Author: Fabian Deutsch \<fabiand@redhat.com\>
+
+## Introduction
+
+Today networking in KubeVirt is not integrated with Kubernetes, a user can
+rather choose to create a specific VM Spec definition to create a particular
+libvirt domxml, to reuse an existing physical device.
+But this is obviously suboptimal
+
+- It is not integrated with Kubernetes
+- It does not provide connectivity with pods
+- It requires to specify host specific bits (the network device to use)
+
+The purpose of this document is to describe one approach of how we can provide
+a minimal integrated solution, with few workarounds.
+
+**Note:** This approach is rough, and not optimized, but the rule of thumb was
+to find a robust solution which wrks with Kubernetes today.
+
+
+## Use-case
+
+The primary use-case is to allow VM to get connected to the pod network.
+
+
+## API
+
+The API is pretty clear: As usual an interface will be defined.
+To indicate that Pod interfaces shall be used for networking, a new type
+`pod-network` is introduced.
+All other fields, concerning the properties of the virtual NIC, can be used
+as usual.
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+      - type: pod-network
+        mac:
+          address: 00:11:22:33:44:55
+        model:
+          type: virtio
+        source:
+          network: default
+```
+
+The `source` field can be omitted initially, as Kubernetes only provides a
+single network for Pods today.
+In future we might want to allow it to select the Pod network to connect to.
+
+
+## Implementation
+
+Conceptually the implementation works by assuming that every vNIC will be
+mapped to a Pod interface. To achieve this, for every vNIC a _new_ Pod
+interface is requested through CNI.
+
+**Note:** The original Pod interface (eth0) is never touched and can be used to
+access libvirtd (thorugh the libvirtd Pod IP) or to provide VM centric services
+through the VM Pod IP.
```

Comment:
We've got 2 pods in play here - the libvirtd POD and the VM POD.  IIUC, you're saying that *neither* of these PODs will have their eth0 touched. ie, the interface provided to PODs by k8s, will only ever be used for host side mgmt access.  Virtual NICs will always be using completely separate interfaces, that are acquired out of band from POD creation.

I'd suggest the term 'Pod interface' be used only when talking about the NIC that k8s provides to a POD by default.

For talking about the NICs we acquire to use for QEMU virtual NIC backend, we should use the term  "VM interface', to make the distinction clear